test_runner/src/main/kotlin/ftl/shard/TestMethodDuration.kt

```diff
@@ -0,0 +1,35 @@
+package ftl.shard
+
+import ftl.args.AndroidArgs
+import ftl.args.IArgs
+import ftl.reports.xml.model.JUnitTestCase
+import ftl.reports.xml.model.JUnitTestResult
+
+fun createTestMethodDurationMap(junitResult: JUnitTestResult, args: IArgs): Map<String, Double> {
+    val junitMap = mutableMapOf<String, Double>()
+
+    // Create a map with information from previous junit run
+
+    junitResult.testsuites?.forEach { testsuite ->
+        testsuite.testcases?.forEach { testcase ->
+            if (!testcase.empty() && testcase.time != null) {
+                val key = if (args is AndroidArgs) testcase.androidKey() else testcase.iosKey()
+                val time = testcase.time.toDouble()
+                if (time >= 0) junitMap[key] = time
+            }
+        }
+    }
+
+    return junitMap
```

Comment:
I think we could return here instance of read only map to avoid potential, unwanted operations on returned object.